src/backend/distributed/utils/colocation_utils.c

```diff
@@ -96,6 +98,61 @@ mark_tables_colocated(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * update_colocation updates the colocation of a table.
+ * if colocate_with -> 'none' then the table is assigned a new
+ * colocation group.
+ */
+Datum
+update_colocation(PG_FUNCTION_ARGS)
+{
+	Oid destinationRelationId = PG_GETARG_OID(0);
+	text *colocateWithTableNameText = PG_GETARG_TEXT_P(1);
+
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+	EnsureTableOwner(destinationRelationId);
+
+	char *colocateWithTableName = text_to_cstring(colocateWithTableNameText);
+	if (pg_strncasecmp(colocateWithTableName, "none", NAMEDATALEN) == 0)
+	{
+		BreakColocation(destinationRelationId);
```

Comment:
We should check the input before doing this (e.g. reject reference table, local table)